samples/extensions/extended_dynamic_state2/extended_dynamic_state2.cpp

```diff
@@ -0,0 +1,1021 @@
+
+	write_descriptor_sets = {
+	    vkb::initializers::write_descriptor_set(
+	        descriptor_sets.background,
+	        VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER,
+	        0,
+	        &matrix_background_buffer_descriptor),
+	    vkb::initializers::write_descriptor_set(
+	        descriptor_sets.background,
+	        VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER,
+	        1,
+	        &background_image_descriptor)};
+
+	vkUpdateDescriptorSets(get_device().get_handle(), static_cast<uint32_t>(write_descriptor_sets.size()), write_descriptor_sets.data(), 0, NULL);
+}
+
+/**
+ * @fn void ExtendedDynamicState2::request_gpu_features(vkb::PhysicalDevice &gpu)
+ * @brief Enabling features related to Vulkan extensions
+ */
+void ExtendedDynamicState2::request_gpu_features(vkb::PhysicalDevice &gpu)
+{
+	/* Enable extension features required by this sample
+	   These are passed to device creation via a pNext structure chain */
+	auto &requested_extended_dynamic_state2_features                                   = gpu.request_extension_features<VkPhysicalDeviceExtendedDynamicState2FeaturesEXT>(VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_EXTENDED_DYNAMIC_STATE_2_FEATURES_EXT);
+	requested_extended_dynamic_state2_features.extendedDynamicState2                   = VK_TRUE;
+	requested_extended_dynamic_state2_features.extendedDynamicState2PatchControlPoints = VK_TRUE;
+
+	auto &requested_extended_dynamic_state_feature                = gpu.request_extension_features<VkPhysicalDeviceExtendedDynamicStateFeaturesEXT>(VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_EXTENDED_DYNAMIC_STATE_FEATURES_EXT);
+	requested_extended_dynamic_state_feature.extendedDynamicState = VK_TRUE;
+
+	// Tessellation shader support is required for this example
+	auto &requested_features = gpu.get_mutable_requested_features();
+	if (gpu.get_features().tessellationShader)
+	{
+		requested_features.tessellationShader = VK_TRUE;
+	}
+	else
+	{
+		throw vkb::VulkanException(VK_ERROR_FEATURE_NOT_PRESENT, "Selected GPU does not support tessellation shaders!");
+	}
+
+	if (gpu.get_features().fillModeNonSolid)
+	{
+		requested_features.fillModeNonSolid = VK_TRUE;
+	}
+
+	if (gpu.get_features().samplerAnisotropy)
+	{
+		gpu.get_mutable_requested_features().samplerAnisotropy = true;
+	}
+}
+
+/**
+ * @fn void ExtendedDynamicState2::on_update_ui_overlay(vkb::Drawer &drawer)
+ * @brief Projecting GUI and transferring data between GUI and app
+ */
+void ExtendedDynamicState2::on_update_ui_overlay(vkb::Drawer &drawer)
+{
+	if (drawer.header("Settings"))
+	{
+		if (drawer.checkbox("Tessellation Enable", &gui_settings.tessellation))
+		{
+			update_uniform_buffers();
+		}
+
+		/* Maximum tessellation factor is set to 4.0 */
+		if (drawer.slider_float("Tessellation Factor", &gui_settings.tess_factor, 1.0f, 4.0f))
+		{
+			update_uniform_buffers();
+		}
+	}
+	if (drawer.header("Models"))
+	{
+		drawer.checkbox("Selection effect active", &gui_settings.selection_active);
+		int                      obj_cnt = scene_elements_baseline.size();
```

Comment:
compiler warning on conversion from `size_t` to `int` -> use a `static_cast<int>` or change `obj_cnt` to a a `size_t` (and then the loop variable `i` below as well).

Reply:
fixed, static_cast used